webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/index.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useCallback, useMemo} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import ChevronDownIcon from '@mattermost/compass-icons/components/chevron-down';
+import type {SchedulingInfo} from '@mattermost/types/schedule_post';
+
+import * as Menu from 'components/menu';
+import Timestamp from 'components/timestamp';
+
+import './style.scss';
+
+type Props = {
+    disabled?: boolean;
+    onSelect: (e: React.FormEvent, schedulingInfo: SchedulingInfo) => void;
+}
+
+export function SendPostOptions({disabled, onSelect}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleOnSelect = useCallback((e: React.FormEvent, scheduledAt: number) => {
+        const schedulingInfo: SchedulingInfo = {
+            scheduled_at: scheduledAt,
+        };
+
+        onSelect(e, schedulingInfo);
+    }, [onSelect]);
+
+    const coreMenuOptions = useMemo(() => {
+        const today = new Date();
+        today.setHours(9, 0, 0, 0); // 9 AM
+        const unixTimestamp = today.getTime();
+
+        const timeComponent = (
+            <Timestamp
+                value={unixTimestamp}
+                useDate={false}
+            />
+        );
+
+        const tomorrow9amTime = new Date();
+        tomorrow9amTime.setHours(9, 0, 0, 0);
+        tomorrow9amTime.setDate(today.getDate() + 1);
```

Comment:
I think we might want to do the `setDate` first because of daylight savings time (the programmer's worst enemy)

Reply:
Ah thanks! You saved a couple of hours for future me 🙏